presto-main/src/main/java/io/prestosql/connector/system/jdbc/TableJdbcTable.java

```diff
@@ -85,9 +86,14 @@ public RecordCursor cursor(ConnectorTransactionHandle transactionHandle, Connect
         Optional<String> tableFilter = tryGetSingleVarcharValue(constraint, 2);
         Optional<String> typeFilter = tryGetSingleVarcharValue(constraint, 3);
 
+        Builder table = InMemoryRecordSet.builder(METADATA);
+        if (!isLowerCase(schemaFilter) || !isLowerCase(tableFilter)) {
+            // Uppercase predicate will never match a lowercase name
+            return table.build().cursor();
```

Comment:
This is something to be removed in https://github.com/prestosql/presto/issues/17

If we place the logic closer to where it's needed (eg when QualifiedTablePrefix is constructed), then 

- it would be more obvious why the check matters
- it would be easier to track it down and remove when doing above issue